rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+      summary: Drop a namespace from the catalog. Namespace must be empty.
+      operationId: dropNamespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: boolean
+        # TODO - Not empty exception, No such namespace exception
+        '401':
+          description: Unauthorized
+components:
+  schemas:
+    CommitTableRequest:
+      type: object
+      properties:
+        tableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        metadataJson:
+          type: string
+    TableIdentifier:
+      type: object
+      required:
+        - namespace
+      properties:
+        namespace:
+          type: array
+          items:
+            type: string
+          nullable: false  # Note this is ignored
+        name:
+          pattern: \S
+          type: string
+          nullable: false
+    CommitTableResponse:
+      type: object
+      properties:
+        metadataLocation:
+          type: string
+        metadataJson:
+          type: string
+    RemovePropertiesRequest:
+      type: object
+      properties:
+        namespace:
+          type: string
+        properties:
+          uniqueItems: true
+          type: array
+          items:
+            type: string
+    Catalog:
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+          description: Unique identifier for this catalog
+        name:
+          type: string
+        location:
+          type: string
+          description: Warehouse location for this catalog or URI of metastore or other identifying location
+        properties:
+          type: object
+          description: Additional catalog level properties
+          default: {}
+      required:
+        - name
+        - properties
+    CreateNamespaceRequest:
+      type: object
+      properties:
+        namespace:
+          type: array
+          description: individual levels of the namespace
+          items:
+            type: string
+        properties:
+          type: object
+          description: Configuration properties for the namespace
+          example: '{ owner: "Hank Bendickson" }'
+    RenameTableRequest:
+      type: object
+      properties:
+        sourceTableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        destinationTableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+    CreateTableRequest:
+      type: object
+      properties:
+        identifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        schema:
+          $ref: '#/components/schemas/Schema'
+        partitionSpec:
+          $ref: '#/components/schemas/PartitionSpec'
+        sortOrder:
+          $ref: '#/components/schemas/SortOrder'
+        properties:
+          type: object
+          additionalProperties:
+            type: string
+        metadataJson:
```

Comment:
I don't think this should be in the request, since it could differ from the individual objects (e.g., schema).